src/sentry/taskworker/pending_task_store.py

```diff
@@ -33,7 +33,7 @@ def get_pending_task(self) -> InflightActivation | None:
                 return None
 
             # TODO this duration should be a tasknamespace setting, or with an option
-            deadline = task.added_at + timedelta(minutes=3)
+            deadline = datetime.now() + timedelta(minutes=3)
```

Comment:
The processing deadline should be the timestamp at which the task was pulled out of the datastore to be sent/received to/by the worker plus some adjustable duration. The `added_at` timestamp is first captured when the message was read from kafka and inserted to the datastore. In the scenario where the worker does not pick up the task before `task.added_at + <adjustable duration>`, then the task can never be completed. More simply, the current time will always be ahead of processing_deadline.